app/models/spree_multi_vendor/spree/order_decorator.rb

```diff
@@ -1,8 +1,12 @@
 module SpreeMultiVendor::Spree::OrderDecorator
   def self.prepended(base)
     base.has_many :commissions, class_name: 'Spree::OrderCommission'
-    base.state_machine.after_transition to: :complete, do: :generate_order_commissions
-    base.state_machine.after_transition to: :complete, do: :send_notification_mails_to_vendors
+  end
+
+  def finalize!
```

Comment:
@hoshinotsuyoshi I think we can avoid this by changing

```ruby
Spree::Order.prepend SpreeMultiVendor::Spree::OrderDecorator
```

to 

```ruby
Spree::Order.prepend SpreeMultiVendor::Spree::OrderDecorator unless Spree::Order.included_modules.include?(SpreeMultiVendor::Spree::OrderDecorator)

Reply:
👋  Thank you for your reviewing. 

OK.  I added commit about this. 1e30b4275fef436f4be654272b686cc71a4beaa

I think such a method was recently added to the guide.  :+1: https://github.com/spree/spree/commit/9ff42905579e2fda1c760fca528aae3aa822e29a

Reply:
@hoshinotsuyoshi so with this approach we could skip the `finalize!` overwrite altogether

Reply:
Ah, I may have misunderstood your intentions. 🙏 

Let me check about `unless Spree::Order.included_modules.include?(SpreeMultiVendor::Spree::OrderDecorator)` behavior whether is enough or not.   

I'll tackle this  on Friday or Saturday.
Stay tune!

Reply:
I confirmed the behavior. It seems to be OK. 

rebased && squashed && force-pushed. 

https://github.com/spree-contrib/spree_multi_vendor/pull/172/commits/e1879a22c74fed8d079eb08d156cb532b91f7206 
https://github.com/spree-contrib/spree_multi_vendor/pull/172/commits/a5333cfa0ac21010f1693f121a009c10f5d360bd

And edited PR description. https://github.com/spree-contrib/spree_multi_vendor/pull/172#issue-586246938

Thank you! @damianlegawiec